homeassistant/components/alarm_control_panel/device_trigger.py

```diff
@@ -0,0 +1,104 @@
+"""Provides device automations for Alarm control panel."""
+from typing import List
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_DOMAIN,
+    CONF_TYPE,
+    CONF_PLATFORM,
+    CONF_DEVICE_ID,
+    CONF_ENTITY_ID,
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_PENDING,
+    STATE_ALARM_TRIGGERED,
+)
+from homeassistant.core import HomeAssistant, CALLBACK_TYPE
+from homeassistant.helpers import config_validation as cv, entity_registry
+from homeassistant.helpers.typing import ConfigType
+from homeassistant.components.automation import state, AutomationActionType
+from homeassistant.components.device_automation import TRIGGER_BASE_SCHEMA
+from . import DOMAIN
+
+TRIGGER_TYPES = {
+    "pending",
+    "triggered",
+    "disarmed",
+    "armed_home",
+    "armed_away",
+    "armed_night",
+}
+
+TRIGGER_SCHEMA = TRIGGER_BASE_SCHEMA.extend(
+    {
+        vol.Required(CONF_ENTITY_ID): cv.entity_id,
+        vol.Required(CONF_TYPE): vol.In(TRIGGER_TYPES),
+    }
+)
+
+
+async def async_get_triggers(hass: HomeAssistant, device_id: str) -> List[dict]:
+    """List device triggers for Alarm control panel devices."""
+    registry = await entity_registry.async_get_registry(hass)
+    triggers = []
+
+    # Get all the integrations entities for this device
+    for entry in entity_registry.async_entries_for_device(registry, device_id):
+        if entry.domain != DOMAIN:
+            continue
+
+        # Add triggers for each entity that belongs to this integration
+        for trigger_type in TRIGGER_TYPES:
```

Comment:
Please use the new supported features introduced in #29065 to make sure the device support specific triggers.

Reply:
So just append the current trigger if the following passes?
```python
if trigger_type in entry.supported_features:
```

Reply:
that's not how supported features works. Supported features is a bit map. See the linked PR on how to check if a feature is supported.

Reply:
I could only find the following example, is that the correct way to use it?
https://github.com/home-assistant/home-assistant/blob/1c73ac5df88a969c622bd48c788c7fed0d4fbd59/homeassistant/components/alarm_control_panel/device_action.py#L62-L70

Reply:
yes

Reply:
Okay, thank you very much 😊

Reply:
Implemented with https://github.com/home-assistant/home-assistant/pull/29068/commits/9d72a4b5687f19cf23ec4983a0a44e3c99e20518 :)